### PR TITLE
Allow JSON 2.x

### PR DESCRIPTION
**Pull request by jkeiser**
As Chef and ChefDK move to Ruby 2.3 (which is happening as of this next release), we're moving away from Ruby 2.1 support. This means that the JSON version can finally be bumped. In the meantime it's best to allow both the old and new version (which allows us to upgrade when ready).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ Berksfile.lock
 .librarian
 Puppetfile.lock
 .kitchen.local.yml
+binstubs
```
Comment (chris-rock) [inline .gitignore:12]: why do we need this here?

File: .travis.yml [hunk 2]
```diff
@@ -17,9 +17,9 @@ before_install:
 
 matrix:
   include:
-  - rvm: 1.9.3
-  - rvm: 2.0
-  - rvm: 2.2
+  - rvm: 2.2.5
```
Comment (chris-rock) [inline .travis.yml:20]: At the moment, most workstations use `gem install inspec`, therefore we need to support more versions that chef. Therefore we should not change the old version behavior.

File: inspec.gemspec [hunk 3]
```diff
@@ -26,7 +26,7 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency 'r-train', '~> 0.10.4'
   spec.add_dependency 'thor', '~> 0.19'
-  spec.add_dependency 'json', '~> 1.8'
+  spec.add_dependency 'json', '>= 1.8', '< 3.0'
```
Comment (chris-rock) [inline inspec.gemspec:29]: 👍

File: appveyor.yml [hunk 4]
```diff
@@ -4,8 +4,8 @@ platform:
 
 environment:
   matrix:
-    - ruby_version: "200-x64"
-    - ruby_version: "21"
+    - ruby_version: "23-x64"
```
Comment (chris-rock) [inline appveyor.yml:7]: as described above, I do not see the need to change it here

**Comment by chris-rock** [comment]
@jkeiser Thanks for adding JSON 2 support 👍 Can we focus the PR on only this one improvement please?

**Comment by jkeiser** [comment]
@chris-rock we can't focus on just json 2.0, because once you allow 2.0 you become 2.1-incompatible.

**Comment by jkeiser** [comment]
The reason to change things to Ruby 2.3 is, the major delivery vehicle for inspec (chef and chef-dk) is on Ruby 2.3. It seems prudent to test against the Ruby you will primarily run on, is all. Happy to change it to 2.2, if you so desire, but I can't change it to 2.1 with the relaxing of the constraint.

File: .gitignore (hunk 1 above)
Comment (jkeiser) [reply .gitignore:12]: It's a typical place where `bundle install` puts binstubs (the main alternative to `bin`, which we use here).

File: .travis.yml (hunk 2 above)
Comment (jkeiser) [reply .travis.yml:20]: If we can't abandon 2.1, then we can't do this change at all. Note that Chef and Chef DK have abandoned 2.1 (as of the next release) and in fact will not `gem install` on those versions anymore without a `Gemfile`.

File: inspec.gemspec (hunk 3 above)
Comment (jkeiser) [reply inspec.gemspec:29]: json 2.0 requires ruby version 2.2, and a gem install doesn't take this into account (sadly), so relaxing this constraint will break gem installs on Ruby 2.1 and below.

**Comment by jkeiser** [comment]
@chris-rock in any case, the tests are failing on 2.3 in ways that make me wonder whether it even _works_ on 2.3. It would be prudent to test `inspec` on 2.3 before the next `chef` release, regardless of whether we make the leap to JSON 2.0.

**Comment by jkeiser** [comment]
@chris-rock FYI, I created https://github.com/chef/inspec/pull/971 to add Ruby 2.3 to the test matrix. Let's revisit this once that's all worked out.

**Pull request closed without merging**